crates/turborepo-filewatch/src/hash_watcher.rs

```diff
@@ -1285,4 +1285,78 @@ mod tests {
         )
         .await;
     }
+
+    #[tokio::test]
+    #[tracing_test::traced_test]
+    async fn test_negative_inputs() {
+        let (_tmp, _repo, repo_root) = setup_fixture();
+
+        let watcher = FileSystemWatcher::new_with_default_cookie_dir(&repo_root).unwrap();
+
+        let recv = watcher.watch();
+        let cookie_writer = CookieWriter::new(
+            watcher.cookie_dir(),
+            Duration::from_millis(100),
+            recv.clone(),
+        );
+
+        let scm = SCM::new(&repo_root);
+        assert!(!scm.is_manual());
+        let package_watcher = PackageWatcher::new(repo_root.clone(), recv, cookie_writer).unwrap();
+        let package_discovery = package_watcher.watch_discovery();
+        let hash_watcher =
+            HashWatcher::new(repo_root.clone(), package_discovery, watcher.watch(), scm);
+
+        let foo_path = repo_root.join_components(&["packages", "foo"]);
+        let dist_path = foo_path.join_component("dist");
+        dist_path
+            .join_component("some-dist-file")
+            .create_with_contents("dist file")
+            .unwrap();
+        dist_path
```

Comment:
If I'm reading this correctly, this shouldn't get matched regardless, since we're only watching files ending in `-file`, no? Should we rename this to `extra-file`?

Reply:
Good call, changed.